src/client/app/reducers/admin.js

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as adminActions from '../actions/admin';
+import { chartTypes } from './graph';
+
+const defaultState = {
+	displayTitle: '',
+	defaultChartToRender: chartTypes.line,
+	defaultBarStacking: false,
+	isFetching: false,
+	isSubmitting: false
+};
+
+export default function admin(state = defaultState, action) {
+	switch (action.type) {
+		case adminActions.UPDATE_DISPLAY_TITLE:
+			return {
+				...state,
+				displayTitle: action.displayTitle
+			};
+		case adminActions.UPDATE_DEFAULT_CHART_TO_RENDER:
+			return {
+				...state,
+				defaultChartToRender: action.defaultChartToRender
+			};
+		case adminActions.TOGGLE_DEFAULT_BAR_STACKING:
+			return {
+				...state,
+				defaultBarStacking: !state.defaultBarStacking
+			};
+		case adminActions.REQUEST_PREFERENCES:
+			return {
+				...state,
+				isFetching: true
+			};
+		case adminActions.RECEIVE_PREFERENCES:
+			return {
+				...state,
+				isFetching: false,
+				displayTitle: action.data.displayTitle,
+				defaultChartToRender: action.data.defaultChartToRender,
+				defaultBarStacking: action.data.defaultBarStacking
+			};
+		case adminActions.TOGGLE_IS_SUBMITTING_PREFERENCES:
```

Comment:
I think it's better to have two functions, one to set `submitted = true` the other to set it to `false`. Doing that makes it much harder to toggle `submitted` into the wrong state.